### PR TITLE
Milestone 4

### PR DESCRIPTION
**Pull request by Guy-Gustave**
Milestone 4

In this milestone, I added logic to the calculator, by creating another folder "logic" which contains 2 JS files

### CODE REVIEW

File: src/logic/calculatate.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import operate from './operate';
+
+export default function calculatate(calculator, buttonName) {
+  let { total, next, operation } = calculator;
+
+  const operations = ['+', '-', 'x', '÷', '%'];
+  // const digits = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9'];
+  // eslint-disable-next-line no-unused-vars
+  // const operationTotal = null;
```
Comment (addod19) [inline src/logic/calculatate.js]: Kindly remove all commented code, use git history instead :+1:

File: src/logic/operate.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    case 'x':
+      return x.mul(y).toString();
+    default:
+      if (numberTwo === '0') {
+        total = 'Infinity';
+      } else {
+        total = x.div(numberTwo).toString();
+      }
+      return total;
+  }
```
Comment (addod19) [inline src/logic/operate.js:23]: Kindly add a case for a modifier operation where we turn numbers into percentages :+1:

File: src/logic/operate.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+      return x.minus(y).toString();
+    case '+':
+      return x.plus(y).toString();
+    case 'x':
+      return x.mul(y).toString();
+    default:
+      if (numberTwo === '0') {
+        total = 'Infinity';
+      } else {
+        total = x.div(numberTwo).toString();
```
Comment (addod19) [inline src/logic/operate.js]: Kindly change the `numberTwo` inside the "div()" to "y" instead since "y" is already wrapped with the Big class

**Review by addod19 — CHANGES_REQUESTED** [review]
## Status: Require changes :recycle: :traffic_light: 

Hi @Guy-Gustave, congratulations on this milestone :+1: The README file is professional :smile: The calculate.js is imported in the App component :1st_place_medal: The calculate.js module is implemented as expected :100: However, there are a few comments we need to address before approving the milestone :+1: Kindly check the comments below:

- Kindly edit the PR so we have a title and a short description of the milestone :+1: 
- Kindly add the modifier operation so we can convert numbers to percentages :1st_place_medal: 

Once these are fixed, kindly submit another code review request :smile: 
_Happy coding_ :man_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

**Review by arslanbisharat — CHANGES_REQUESTED** [review]
### Status: Require changes ♻️ 🚥

Hi @Guy-Gustave,
Congratulations on this milestone. 👍 The README file is professional 😄 The calculate.js is imported in the App component 🥇 The calculate.js module is implemented as expected 💯 , However, there are a few comments we need to address before approving the milestone 👍 Kindly check the comments below:

- Kindly edit the PR so we have a title and a short description of the milestone 👍

- Kindly add the modifier operation so we can convert numbers to percentages 🥇
- Remove comments from your code as mentioned by the previous TSE. 


Once these are fixed, kindly submit another code review request 😄
Happy coding 👨‍💻

Best,
Arslan Bisharat

**Comment by Guy-Gustave** [comment]
this is a pull request of milestone 4 I made after reviewing bye a Tse. I did all corrections suggested and the review was helpful

**Comment by Guy-Gustave** [comment]
the corrections are done now. sorry for it.

**Review by addod19 — CHANGES_REQUESTED** [review]
## Status: Require changes :recycle: :traffic_light: 

Hi @Guy-Gustave, congratulations for editing the PR and also adding the percentage functionality :fireworks: However, as mentioned in the previous reviews, we were asked the to make changes in the following:
- Instead of using the `div` for converting numbers into percentages we are required to use the `mod` instead
- We were also asked to remove comments from the calculate.js file but we still have those comments

[Optional]
- It would be great if we convert our functional components into ES 6 functions

Once these are fixed, kindly submit another code review request :smile: 
_Happy coding_ :man_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

**Review by Torres-ssf — APPROVED** [review]
### PR Approved :+1: 
Hi @Guy-Gustave :hand: 
Great job on this PR.
Good luck with the next one

Happy coding :computer: 
**Sérgio Torres**

**Pull request merged**